### PR TITLE
[API] Introducing remote debugging

### PR DESCRIPTION
**Pull request by liranbg**
Instructions is written on `CONTRIBUTING.md` file

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -119,3 +119,41 @@ Currently, this can only be done by one of the maintainers, the process is:
 (leave all options default)
 3. Go to the [system test action](https://github.com/mlrun/mlrun/actions?query=workflow%3A%22System+Tests%22) and trigger 
 it for the branch, change "Take tested code from action REF" to `true`   
+
```
Comment (Tankilevitch) [inline CONTRIBUTING.md:122]: Lets move it to the new `docs/developer` directory.

File: mlrun/api/main.py [hunk 2]
```diff
@@ -56,6 +57,21 @@
 app.include_router(api_router, prefix=API_PREFIX, include_in_schema=False)
 
 
+def start_debugging() -> None:
+    if "enabled" not in os.environ.get("MLRUN_API_DEBUG_MODE", "").lower():
```
Comment (Tankilevitch) [inline mlrun/api/main.py:61]: Why this is not part of mlrun config object?

File: .github/workflows/build-workflow-matrix.json [hunk 3]
```diff
@@ -5,6 +5,9 @@
   {
     "image-name":"api"
   },
+  {
+    "image-name": "api-dev"
+  }
```
Comment (Tankilevitch) [inline .github/workflows/build-workflow-matrix.json:10]: I wonder whether we want to build that image on every release, or just when building the merge builds, something like we are doing in `- name: Pull cache, build and push test image` step in build.yaml

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good, minor comments

File: mlrun/api/main.py [hunk 4]
```diff
@@ -56,6 +57,21 @@
...
+        return
+
+    # pydevd_pycharm is installed only on development image. do not move this import to top
+    import pydevd_pycharm
+
+    pydevd_pycharm.settrace(
+        os.environ.get("MLRUN_API_DEBUG_HOST", "localhost"),
+        port=int(os.environ.get("MLRUN_API_DEBUG_PORT", 40000)),
```
Comment (quaark) [inline mlrun/api/main.py:69]: Same thing regarding @Tankilevitch's comment above. These should probably be in the config in a `debug` section

**Review by quaark — CHANGES_REQUESTED** [review]
Looks great! just a minor thing adding to @Tankilevitch's comments

File: mlrun/config.py [hunk 5]
```diff
@@ -186,7 +186,15 @@
...
-        "debug": False,
+        "debug": {
+            "enabled": False,
+            # when enabled, pydevd-pycharm will be started trying to connect to the host & port specified
+            "pydevd_pycharm": {
+                "enabled": False,
+                "port": 4000,
+                "host": "localhost",
+            },
+        },
```
Comment (Tankilevitch) [inline mlrun/config.py:197]:
```suggestion
        "debug": {
            "mode": "disabled",
            # when enabled, pydevd-pycharm will be started trying to connect to the host & port specified
            "pydevd_pycharm": {
                "mode": "disabled",
                "port": 4000,
                "host": "localhost",
            },
        },
```
I think that will allow us more room to grow if in the future we may want to add another behaviors to the debug modes.

File: Makefile [hunk 6]
```diff
@@ -325,6 +325,7 @@ MLRUN_API_IMAGE_DOCKER_CACHE_FROM_FLAG := $(if $(and $(MLRUN_DOCKER_CACHE_FROM_T
 MLRUN_API_CACHE_IMAGE_PULL_COMMAND := $(if $(and $(MLRUN_DOCKER_CACHE_FROM_TAG),$(MLRUN_USE_CACHE)),docker pull $(MLRUN_API_CACHE_IMAGE_NAME_TAGGED) || true,)
 MLRUN_API_CACHE_IMAGE_PUSH_COMMAND := $(if $(and $(MLRUN_DOCKER_CACHE_FROM_TAG),$(MLRUN_PUSH_DOCKER_CACHE_IMAGE)),docker tag $(MLRUN_API_IMAGE_NAME_TAGGED) $(MLRUN_API_CACHE_IMAGE_NAME_TAGGED) && docker push $(MLRUN_API_CACHE_IMAGE_NAME_TAGGED),)
 DEFAULT_IMAGES += $(MLRUN_API_IMAGE_NAME_TAGGED)
+DEFAULT_IMAGES += $(MLRUN_API_DEV_IMAGE_NAME_TAGGED)
```
Comment (Tankilevitch) [inline Makefile:328]: redundant?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good last minor comments

**Comment by liranbg** [comment]
maybe later

**Pull request closed without merging**